src/laser_geometry.cpp

```diff
@@ -720,4 +720,33 @@ const boost::numeric::ublas::matrix<double>& LaserProjection::getUnitVectors_(do
                                     channel_options);
   }
 
+  void LaserProjection::transformLaserScanToPointCloud_ (const std::string &target_frame,
+                                                         const sensor_msgs::LaserScan &scan_in,
+                                                         sensor_msgs::PointCloud2 &cloud_out,
+                                                         const std::string &fixed_frame,
+                                                         tf2::BufferCore &tf,
+                                                         double range_cutoff,
+                                                         int channel_options)
+  {
+    const auto& start_time = scan_in.header.stamp;
+    auto end_time = scan_in.header.stamp;
+    if(!scan_in.ranges.empty()) end_time += ros::Duration ().fromSec ( (scan_in.ranges.size()-1) * scan_in.time_increment);
+
+    const auto start_transform = tf.lookupTransform (target_frame, scan_in.header.frame_id, start_time);
+    const auto end_transform = tf.lookupTransform (target_frame, start_time, scan_in.header.frame_id, end_time, fixed_frame);
```

Comment:
The use of auto for these local variables seems a bit gratuitous to me - I think this would be more readable if the types were clear to the reader. (auto would make more sense if the type were some really long templated iterator thing we don't want to write out)

Reply:
Okay, substituted with full types.